cub/test/catch2_test_device_reduce_nondeterministic.cu

```diff
@@ -82,7 +84,13 @@ C2H_TEST("Nondeterministic Device reduce works with float and double on gpu",
   // TODO: Use std::reduce once we drop support for GCC 7 and 8
   h_expected[0] = std::accumulate(h_input.begin(), h_input.end(), type{}, cuda::std::plus<type>());
 
-  REQUIRE_APPROX_EQ_EPSILON(h_expected, d_output, type{0.01});
+  // relative round-off error of recursive summation is proportional to n * type::epsilon,
+  // see https://epubs.siam.org/doi/epdf/10.1137/19M1257780
+
+  type relative_err = std::min((num_items + 1) * std::numeric_limits<type>::epsilon(), static_cast<type>(1));
```

Comment:
Suggestion: prefer `cuda::std` APIs so we can give them more test coverage. They also work in a few more cases. E.g. `cuda::std::numeric_limits` also works for extended FP types. 
```suggestion
  type relative_err = cuda::std::min((num_items + 1) * cuda::std::numeric_limits<type>::epsilon(), static_cast<type>(1));
```